recipes/cyclonedds/all/conanfile.py

```diff
@@ -0,0 +1,174 @@
+from conan.tools import files, build, scm
+from conan.tools.cmake import CMakeToolchain, CMake, CMakeDeps
+from conan.tools.layout import cmake_layout
+
+required_conan_version = ">=1.43.0"
+
+class CycloneDDSConan(ConanFile):
+    name = "cyclonedds"
+    license = "EPL-2.0"
+    homepage = "https://cyclonedds.io/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Eclipse Cyclone DDS - An implementation"\
+                  " of the OMG Data Distribution Service (DDS) specification"
+    topics = ("dds", "ipc", "ros", "middleware")
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "ssl": [True, False],
+        "shm" : [True, False],
+        "security" : [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "ssl": False,
+        "shm": False,
+        "security": False
+    }
+
+    short_paths = True
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join("lib", "cmake", f"conan-official-{self.name}-targets.cmake")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+    
+    def _create_cmake_module_alias_targets(self, module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        files.save(self, module_file, content)
+
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.shm:
+            self.requires("iceoryx/2.0.0")
+        if self.options.ssl:
+            self.requires("openssl/1.1.1q")
+    
+    def build_requirements(self):
+        self.tool_requires("cmake/3.16.2")
+
+    def validate(self):
+        compiler = self.settings.compiler
+        version = scm.Version(self.settings.compiler.version)
+
+        if ((self.options.security is True ) and (self.options.shared is False)):
+            raise ConanInvalidConfiguration("Cyclone DDS currently do not support"\
+                                            "static build and security on")
+
+        if compiler.get_safe("cppstd"):
+            build.check_min_cppstd(self, 14)
+
+        if compiler == "Visual Studio":
```

Comment:
```suggestion
        if is_msvc(self):
```
https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc